### PR TITLE
Docs: lint profiles

### LINKED ISSUE #2663: Docs: Add details for using profiles
##### Summary

As content moves through the automation content creation pipeline, various linting rules that are developed against may or may not apply depending on where the content is being executed.  

For example, in development, as a content creator, there should be the ability to create some rules and quickly validate them utilising ansible-lint locally.  Then in the first test run, an ansible-lint profile should allow the executor from ignoring the rules to ensure that the content itself runs as desired.

Update documentation to provide context for using profiles in a more progressive (step-by-step) way with additional context that addresses motivation.

##### Issue Type

- Bug Report (Docs)

##### Ansible and Ansible Lint details

N/A

##### OS / ENVIRONMENT

Fedora 36

##### STEPS TO REPRODUCE

N/A

##### Desired Behavior

N/A

##### Actual Behavior

N/A

### PR DESCRIPTION
**Pull request by oraNod**
Fixes #2663 with documentation edits for applying and using lint profiles.

### CODE REVIEW

**Comment by oraNod** [comment]
@cidrblock Hey, hope you're on the mend and recovering from the funk. When you have a moment could you please take a look at this update for lint profiles? We spoke about sort of a "getting started" style of doc for this and I'd like to make sure this hits the mark. Thank you.

File: docs/using-profiles.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+Starting with the minimal profile ensures that Ansible can load your content.
+As you move to the next stage of developing content, you can gradually apply profiles to avoid common pitfalls and brittle complexity.
+Then, when you are ready to publish or share your content, you can use the ``shared`` and ``production`` profiles with much stricter rules that harden security and guarantee reliability.
+
+```{note}
+Tags such as `opt-in` and `experimental` do not take effect for rules that are included in profiles, directly or indirectly.
+If a rule is in a profile, Ansible-lint applies that rule to the content.
+```
+
+After you install and configure `ansible-lint`, you can apply profiles as follows:
```
Comment (oraNod) [inline docs/using-profiles.md:17]: @priyamsahoo and @ganeshrn Is it possible to use profiles on the command line only? I tried to set a ``--profile=foo`` in the vscode extension but it didn't work.

Also I'm curious if it's possible to set a profile in the `.ansible-lint` configuration file. I could imagine a usage scenario where a profile might be imposed.

File: docs/using-profiles.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+(using-lint-profiles)=
+
+# Applying lint profiles
+
+Ansible-lint profiles allow content creators to progressively improve the quality of Ansible playbooks, roles, and collections.
```
Comment (oraNod) [inline docs/using-profiles.md:5]: @ansible/devtools "s/content creator/content developer" ?

Comment (oraNod) [reply docs/using-profiles.md:5]: @ssbarnea not sure if you spotted this or have a strong feeling. I'm leaving as-is to identify the "content creator" role. Cheers.

Comment (ssbarnea) [reply docs/using-profiles.md:5]: Nope, please keep the old URL (filename), I am not keen on altering the URLs as it requires extra work. Better to improve the docs without changing urls.

Comment (oraNod) [reply docs/using-profiles.md:5]: @ssbarnea sorry do you mean `(using-lint-profiles)=` on line 1?

Comment (ssbarnea) [reply docs/using-profiles.md:5]: Yeah. But if you change that one you also need to rename back to `profiles.md` too.

Comment (cidrblock) [reply docs/using-profiles.md:5]: +1 for leaving content creator

File: docs/using-profiles.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
+(using-lint-profiles)=
+
+# Applying profiles
+
+Ansible-lint profiles allow content creators to progressively improve the quality of Ansible playbooks, roles, and collections.
+
+During early development cycles, you need Ansible-lint rules to be less strict.
+Starting with the minimal profile ensures that Ansible can load your content.
+As you move to the next stage of developing content, you can gradually apply profiles to avoid common pitfalls and brittle complexity.
+Then, when you are ready to publish or share your content, you can use the `shared` and `production` profiles with much stricter rules that harden security and guarantee reliability.
```
Comment (cidrblock) [inline docs/using-profiles.md:10]: and guarantee reliability.  The `shared` and `production` profiles exist to ensure your Ansible content is easy to read and consistently formatted, making it easier for other to contribute to it and use. ( <=something like this? Trying to reinforce why these profiles exist)

File: src/ansiblelint/data/profiles.yml [hunk 4]
```diff
@@ -15,9 +15,7 @@ min:
     syntax-check:
 basic:
   description: >
-    Basic profile enables a set of rules designed to spot very common coding
```
Comment (cidrblock) [inline src/ansiblelint/data/profiles.yml:18]: The basic profile

File: src/ansiblelint/data/profiles.yml [hunk 5]
```diff
@@ -72,13 +67,12 @@ safety:
...
   extends: safety
   description: >
-    The shared profile is for those that want to publish their content, roles
-    or collection to either [galaxy.ansible.com](https://galaxy.ansible.com),
-    [automation-hub](https://console.redhat.com/ansible/automation-hub) or
-    another private instance of these. In addition to all the rules related to
-    packaging and publishing, like metadata checks, this also includes some
-    rules that are known to be good practices for keeping the code readable and
-    maintainable.
+    Ensures that content follows best practices for packaging and publishing.
```
Comment (cidrblock) [inline src/ansiblelint/data/profiles.yml:70]: The shared profile..... (just to be consistent)

File: src/ansiblelint/data/profiles.yml [hunk 6]
```diff
@@ -107,9 +101,9 @@ shared:
       url: https://github.com/ansible/ansible-lint/issues/2038
 production:
   description: >
-    The production profile represents the top level of verification and it
-    is required for inclusion in [Ansible Automation Platform (AAP)](https://www.redhat.com/en/technologies/management/ansible)
-    as either validated or certified content.
+    Ensures that content meets requirements for
```
Comment (cidrblock) [inline src/ansiblelint/data/profiles.yml:104]: The production profile...

**Review by cidrblock — APPROVED** [review]
some suggestions, take or leave it :) good stuff

File: docs/using-profiles.md (hunk 2 above)
Comment (oraNod) [reply docs/using-profiles.md:5]: @ssbarnea Yeah but `profiles.md` is generated so I thought `using-lint-profiles` would be good. I believe that would be the best choice for SEO.

"using-lint-profiles" is slightly more meaningful than "using-profiles"

File: docs/using-profiles.md (hunk 3 above)
Comment (oraNod) [reply docs/using-profiles.md:10]: @cidrblock "These profiles harden security, guarantee reliability, and ensure your Ansible content is easy for others to contribute to and use."

Does that sound reasonable or should I drop the "harden security" and "guarantee reliability"?

**Review by ssbarnea — APPROVED** [review]

**Pull request merged**